src/ophyd_async/core/_detector.py

```diff
@@ -79,20 +79,22 @@ class TriggerInfo(BaseModel):
     frame_timeout: float | None = Field(default=None, gt=0)
     """What is the maximum timeout on waiting for a frame"""
 
-    multiplier: int = 1
-    """How many triggers make up a single StreamDatum index, to allow multiple frames
-    from a faster detector to be zipped with a single frame from a slow detector
-    e.g. if num=10 and multiplier=5 then the detector will take 10 frames,
-    but publish 2 indices, and describe() will show a shape of (5, h, w)
+    frames_per_event: PositiveInt = 1
+    """The number of triggers that are grouped into a single StreamDatum index.
+    A frames_per_event > 1 can be useful to have frames from a faster detector
+    able to be zipped with a single frame from a slower detector. E.g. if
+    number_of_triggers=10 and frames_per_event=5 then the detector will take
+    10 frames, but publish 2 StreamDatum indices, and describe() will show a
+    shape of (5, h, w) for each.
     """
 
     @computed_field
     @cached_property
     def total_number_of_triggers(self) -> int:
         return (
-            sum(self.number_of_triggers)
+            sum(self.number_of_triggers) * self.frames_per_event
```

Comment:
Either we rename `number_of_triggers` to `number_of_events` and this logic is valid, or we remove the `* self.frames_per_event`.
At the moment we have:
- number_of_triggers
- livetime and deadtime of each trigger
- multiplier (which is effectively triggers_per_event)

We could move to:
- number_of_events
- livetime and deadtime of each frame
- frames_per_event

@jwlodek do you have an opinion?

Reply:
I think it should be:

`TriggerInfo.number_of_triggers` -> `TriggerInfo.number_of_events`
`TriggerInfo.frames_per_event` -> `TriggerInfo.triggers_per_event`

This would make the `total_number_of_triggers` property meaningful. The word "frame" is really only relevant to detectors that capture 2-D data.

Then we can add that the word "event" here relates to a single `StreamDatum` index being emitted.

Reply:
I'm happy with that